src/arbitraries/UniqueArbitrary.ts

```diff
@@ -25,6 +28,15 @@ export class UniqueArbitrary<A> extends WrappedArbitrary<A> {
     return Array.from(result.values())
   }
 
+  sampleWithBias(sampleSize = 10): FluentPick<A>[] {
+    const cornerCases = this.cornerCases()
+
+    if (sampleSize <= cornerCases.length)
+      return this.sample(sampleSize)
+
+    return this.sample(sampleSize, cornerCases)
+  }
```

Comment:
I'm not following the rationale here. It seems we have a method called `sampleWithBias` that only actually _samples with bias_ when the search space is larger than the biased space. Otherwise it will ignore the bias. Am I correct?

Reply:
I believe this to be because _during_ the shrinking process we don't want to keep testing the corner cases over and over, particularly in situations where they've been tested before. For small enough `sampleSizes`, the corner cases can occupy a significant portion of a biased sample. This makes me wonder about a few things:

a. Currently our strategy is to assign a 100% probability of sampling a value in the bias pool; this reminds me of a discussion we had long ago about bias being represented instead by a PMF in the Arbitrary;
b. With this new code, the probability of sampling such value drops from 100% to its true probability when the `sampleSize` is lower than the bias pool; even if we have tested it before;
c. Maybe we could have a more generic mechanism of know if a particular test-case has already been generated and discard it.

Reply:
I have read what you said and I have some questions:

a. What PMF means? Also, what is the difference between this sampleWithBias() function and the one that you have developed that led you to consider that this is a new strategy?
c. What do you mean by test-case? Sampling is done for each arbitrary. Do you mean knowing the values generated for each Arbitrary or for the combination of Arbitraries?

Reply:
This appears as new code in the diff; I didn't realised it was existing code xD Anyway:

a. [Probability Mass Function](https://en.wikipedia.org/wiki/Probability_mass_function); something akin to that could be a more generic way of introducing bias during sampling. We should maybe move this discussion/idea into an issue;
b. Consider that a `UniqueArbitrary` exists so that we don't test duplicate values; but during the shrinking process, new arbitraries are spawned, and so the "memory" of the values that were already generated is lost. But that doesn't mean that we should always discard previously tested values from a "single" arbitrary, because they might contribute to a new "test-case" when considering the "combination" of arbitraries. What I'm wondering is if that shouldn't be accounted for. Again, I would move this discussion into a new Issue so that we can think about it in the future.

Reply:
a. I agree with you and I think such discussion despite making sense to be made is out of the scope of what this PR is really about.
b. Yeah I understand what you mean and I also agree we should move such discussion to a new issue. In fact, storing the values generated for each arbitrary is something that I already considered while developing the code for PR #117.

Reply:
> c. Maybe we could have a more generic mechanism of know if a particular test-case has already been generated and discard it.

Ok, as long as we are not discarding a single value from an Arbitrary but a set of values from a test case. The same value might give different results depending on the values of other arbitraries.

Reply:
Yeah, keeping track of the values used by an arbitrary as well as the values that were combined with it has its complexity. Therefore the need for an issue for this problem.

Reply:
But the issue is due to this PR. So we should fix it in this PR.